jwst/source_catalog/tests/test_source_catalog.py

```diff
@@ -122,3 +122,28 @@ def test_model_without_apcorr_data(nircam_model_without_apcorr):
     step = SourceCatalogStep(save_results=False)
     cat = step.run(nircam_model_without_apcorr)
     assert cat is None
+
+
+def test_input_model_reset(nircam_model):
+    """ Changes to input model data are made in SourceCatalogStep - make sure that
+        these changes are correctly reversed so the input model data/err arrays
+        remain unchanged after processing (to avoid copying datamodel),
+        and that the units are in MJy/sr before and after."""
+
+    original_data = nircam_model.data
```

Comment:
```suggestion
    original_data = nircam_model.data.copy()
```
If you don't copy the data here, it will be a reference  to the model data (i.e., the test below will always pass even if the model data changes).